src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static allAnimals = [];
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.allAnimals.push(this);
+    Animal.alive.push(this);
+  }
+
+  static updateAliveAnimals() {
+    Animal.alive = Animal.allAnimals.filter((animal) => animal.health > 0);
+  }
+
+  checkIfAlive() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+      Animal.allAnimals = Animal.allAnimals.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+      animal.checkIfAlive();
+      Animal.updateAliveAnimals();
```

Comment:
The `bite` method correctly checks if the target animal is an instance of `Herbivore` and is not hidden before applying damage. However, after reducing the health of the animal, you're calling both `animal.checkIfAlive` and `Animal.updateAliveAnimals`. This is redundant and can lead to performance issues because you're filtering the alive animals array twice. You should remove the call to `updateAliveAnimals` here since `checkIfAlive` already handles the removal of dead animals.